modules/provenance.nf

```diff
@@ -0,0 +1,40 @@
+process collect_provenance {
+
+  tag { sample_id }
+
+  executor 'local'
+
+  publishDir "${params.outdir}/provenance_files", pattern: "${sample_id}_*_provenance.yml", mode: 'copy'
+
+  input:
+  tuple val(sample_id), path(provenance_files)
+
+  output:
+  tuple val(sample_id), path("${sample_id}_*_provenance.yml")
+
+  script:
+  """
+  cat ${provenance_files} > ${sample_id}_\$(date +%Y%m%d%H%M%S)_provenance.yml
+  """
+}
+
+process pipeline_provenance {
+
+  tag { pipeline_name + " / " + pipeline_version }
+
+  executor 'local'
+
+  input:
+  tuple val(pipeline_name), val(pipeline_version), val(analysis_start)
+  path(other_provenance)
+
+  output:
+  path("pipeline_provenance.yml")
+
+  script:
+  """
+  cat ${other_provenance} > pipeline_provenance.yml
+  printf -- "- pipeline_name: ${pipeline_name}\\n  pipeline_version: ${pipeline_version}\\n- timestamp_analysis_start: ${analysis_start}\\n" >> pipeline_provenance.yml
```

Comment:
Can you add these keys to the pipeline provenance:

```
nextflow_session_id
nextflow_run_name
```

This is some new info that I've just started collecting recently on some of our pipelines, but I think it will be helpful down the road.

see the tbprofiler provenance example:

https://github.com/BCCDC-PHL/tbprofiler-nf?tab=readme-ov-file#provenance

...and how it's collected:

https://github.com/BCCDC-PHL/tbprofiler-nf/blob/71f22bd3a248c3d85c59d2a051c3f710e147135e/main.nf#L27-L33

Reply:
Done. Will be in next commit.